### PR TITLE
Refactor ClientAuthError for reduced size

### PR DESCRIPTION
**Pull request by tnorling**
Same changes as #6414 but for ClientAuthError

### CODE REVIEW

File: change/@azure-msal-browser-c5e0ae79-ede6-4b7d-bbed-ce6d56dd1244.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Refactor ClientAuthError for reduced size",
```
Comment (tnorling) [inline change/@azure-msal-browser-c5e0ae79-ede6-4b7d-bbed-ce6d56dd1244.json]:
```suggestion
  "comment": "Refactor ClientAuthError for reduced size #6433",
```

File: change/@azure-msal-common-f7b47241-6ade-4ca5-9554-66109a37181e.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Refactor ClientAuthError for reduced size",
```
Comment (tnorling) [inline change/@azure-msal-common-f7b47241-6ade-4ca5-9554-66109a37181e.json]:
```suggestion
  "comment": "Refactor ClientAuthError for reduced size #6433",
```

File: change/@azure-msal-node-63afe666-fcae-412e-9c78-de0f5985a4d4.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Refactor ClientAuthError for reduced size",
```
Comment (tnorling) [inline change/@azure-msal-node-63afe666-fcae-412e-9c78-de0f5985a4d4.json]:
```suggestion
  "comment": "Refactor ClientAuthError for reduced size #6433",
```

**Review by Hudson-QC — APPROVED** [review]

**Review by konstantin-msft — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
Approved, Can we publish the min sizes so we know the gains as you go through these improvement  PRs?

**Comment by tnorling** [comment]
> Approved, Can we publish the min sizes so we know the gains as you go through these improvement PRs?

Rather than doing this manually on a per PR basis I think we should set up some automated tooling to do this for us. But in the interim I'm tracking this on a per release basis. For example our latest release resulted in ~20kb savings

**Comment by sameerag** [comment]
> > Approved, Can we publish the min sizes so we know the gains as you go through these improvement PRs?
> 
> Rather than doing this manually on a per PR basis I think we should set up some automated tooling to do this for us. But in the interim I'm tracking this on a per release basis. For example our latest release resulted in ~20kb savings

I meant automation tool that publishes - just generating a min. May be we can do this when nightly builds are enabled. Thanks.

**Pull request merged**